Source: Ingmar Vanhassel <ingmar@exherbo.org>
Upstream: No
Reason: The default Release & Debug build types add various flags that override our user's settings.
	Added our own build types to avoid this. make picks up ${CFLAGS}, ${CXXFLAGS}, ${LDFLAGS} from the env as usual.
	Based on a debian patch: http://svn.debian.org/viewsvn/pkg-kde/branches/kde4/packages/kdelibs/debian/patches/08_add_debian_build_type.diff?rev=10813&view=auto


From 4303e066fa5c290c4fbb264fd042bd4fd5effc3a Mon Sep 17 00:00:00 2001
From: Ingmar Vanhassel <ingmar@exherbo.org>
Date: Thu, 28 Aug 2008 02:46:56 +0200
Subject: [PATCH] Add Exherbo, ExherboDebug build types, to avoid picking up *FLAGS that override user's settings.

---
 cmake/modules/FindKDE4Internal.cmake |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/cmake/modules/FindKDE4Internal.cmake b/cmake/modules/FindKDE4Internal.cmake
index d8f91bf..cec7bbe 100644
--- a/cmake/modules/FindKDE4Internal.cmake
+++ b/cmake/modules/FindKDE4Internal.cmake
@@ -948,11 +948,15 @@ if (CMAKE_COMPILER_IS_GNUCXX)
    set(CMAKE_CXX_FLAGS_DEBUG          "-g -O2 -fno-reorder-blocks -fno-schedule-insns -fno-inline")
    set(CMAKE_CXX_FLAGS_DEBUGFULL      "-g3 -fno-inline")
    set(CMAKE_CXX_FLAGS_PROFILE        "-g3 -fno-inline -ftest-coverage -fprofile-arcs")
+   set(CMAKE_CXX_FLAGS_EXHERBO        "-DNDEBUG -DQT_NO_DEBUG")
+   set(CMAKE_CXX_FLAGS_EXHERBODEBUG   "")
    set(CMAKE_C_FLAGS_RELWITHDEBINFO   "-O2 -g")
    set(CMAKE_C_FLAGS_RELEASE          "-O2 -DNDEBUG -DQT_NO_DEBUG")
    set(CMAKE_C_FLAGS_DEBUG            "-g -O2 -fno-reorder-blocks -fno-schedule-insns -fno-inline")
    set(CMAKE_C_FLAGS_DEBUGFULL        "-g3 -fno-inline")
    set(CMAKE_C_FLAGS_PROFILE          "-g3 -fno-inline -ftest-coverage -fprofile-arcs")
+   set(CMAKE_C_FLAGS_EXHERBO          "-DNDEBUG -DQT_NO_DEBUG")
+   set(CMAKE_C_FLAGS_EXHERBODEBUG     "")
 
    if (CMAKE_SYSTEM_NAME MATCHES Linux)
      set ( CMAKE_C_FLAGS     "${CMAKE_C_FLAGS} -Wno-long-long -std=iso9899:1990 -Wundef -Wcast-align -Werror-implicit-function-declaration -Wchar-subscripts -Wall -W -Wpointer-arith -Wwrite-strings -Wformat-security -Wmissing-format-attribute -fno-common")
-- 
1.6.0.1

